node/src/node/slave/state/spawning.cpp

```diff
@@ -139,12 +139,12 @@ spawning_t::on_spawn(std::chrono::high_resolution_clock::time_point start) {
 }
 
 void
-spawning_t::on_timeout(const std::error_code& ec) {
+spawning_t::on_timeout(std::unique_ptr<api::cancellation_t>& cancellation, const std::error_code& ec) {
     if (ec) {
         COCAINE_LOG_TRACE(slave->log, "spawn timer has called its completion handler: cancelled");
     } else {
         COCAINE_LOG_ERROR(slave->log, "unable to spawn slave: timeout");
-
+        cancellation->cancel();
```

Comment:
This won't be cancelled if the slave calls `spawning_t::cancel` before timeout reached.